### PR TITLE
Remove unused and duplicated GetEncodingFromEnum

### PR DESCRIPTION
**Pull request by iSazonov**
GetEncodingFromEnum and TextEncodingType is not used.
GetEncodingFromEnum duplicate a code
- in [FileSystemProvider.cs
](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/namespaces/FileSystemProvider.cs#L7528)
- in [Utils.cs](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Utils.cs#L1375)

### CODE REVIEW

**Comment by dantraMSFT** [comment]
Even though the code is not referenced in PowerShell; the change should probably be reviewed See https://msdn.microsoft.com/en-us/library/microsoft.powershell.commands.textencodingtype(v=vs.85).aspx

**Comment by iSazonov** [comment]
@dantraMSFT Good catch! Thanks!  From the docs:
>The members of this class cannot be used directly, nor should this class be used to derive other classes

So it is breaking change but formally.

We have another place with the same duplication:
https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Utils.cs#L1375
and this is only used in one place:
https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs#L1077
We should remove this one too.

(It will be easier for us to implement our Encoding RFC if we do this cleanup code.)

**Comment by dantraMSFT** [comment]
@iSazonov without understanding why these types are causing you issues; my suggestion is for both FileSystemProvider and TranscriptionOption depend on Utils.GetEncodingFromEnum. It doesn't make sense to me to have TranscriptionOption depend on FileSystemProvider.

**Comment by iSazonov** [comment]
@dantraMSFT Thanks! I agree and I've added the appropriate commit.

**Review by dantraMSFT — CHANGES_REQUESTED** [review]
I suggest updating GetEncodingFromEnum in Utils.cs to use the various static Encoding fields, such as Encoding.Unicode instead of constructing instances of the derived class.   All of the referenced derived types are read only and constructing instances of them adds unneeded costs.

**Comment by HemantMahawar** [comment]
@PowerShell/powershell-committee reviewed and agreed that we should NOT be removing public api without any good reason.

**Comment by iSazonov** [comment]
@HemantMahawar Thanks! Whether the committee had considered the causes: 
1. The API don't used internal.
2. The API is duplicated in another two places with non-public API.
3. Docs says that the API not really public "The members of this class cannot be used directly, nor should this class be used to derive other classes".
4. The API looks as temporary, PowerShell engine is not "encoding library" and adds no benefits - anyone can use directly similar encoding APIs from .Net Core.  
5. To implement our future encoding RFC we have to fix all encoding API (it is already obvious that we will have to make any changes) including the "public" API or leave it "as is" with legacy behavior (for what?). It makes sense to minimize the places where changes are needed. After this PR, we'll have only two such API's - in Util.cs and PathUtils.cs - later we can merge them into one and get common encoding behavior throughout all code.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/UtilityCommon.cs [hunk 1]
```diff
@@ -15,103 +15,6 @@
...
-                    break;
-            }
-
-            return encoding;
-        }
-    } // class UtilityCommon
-    /// <summary>
-    /// abc
-    /// </summary>
-    public enum TextEncodingType
```
Comment (lzybkr) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/UtilityCommon.cs]: This api is public and should not be removed.

**Comment by lzybkr** [comment]
The internal api is fine to remove.

The public api should not be removed. It's unfortunate, but removing a public api is a breaking change with existing binaries and scripts, so it's not safe to remove. Despite our intentions and documentation, third party code can still depend on the api and removing it should have a strong justification, which I don't see here.

I suggest:

* Remove the unnecessary public api from our reference assemblies if that is appropriate - then no new code binary code can depend on it
* Mark the api as obsolete.
* Not sure which docs you're referring to - but a public enum is very usable anywhere, so we should fix the docs that say the members cannot be used...

**Comment by iSazonov** [comment]
@lzybkr Thanks! I'd revert the API.
>•Remove the unnecessary public api from our reference assemblies

I don't know how?

>•Mark the api as obsolete

Place a comment in code?

>•Not sure which docs you're referring to - but a public enum is very usable anywhere, so we should fix the docs that say the members cannot be used

The docs is ready https://msdn.microsoft.com/en-us/library/microsoft.powershell.commands.textencodingtype%28v=vs.85%29.aspx

**Review by dantraMSFT — CHANGES_REQUESTED** [review]
In the default path for GetEncodingFromEnum; you can simply break; you're already assigning a default at the top of the method. I would leave the comment there, though.

**Comment by iSazonov** [comment]
Default path is fixed.

**Review by dantraMSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
@dantraMSFT @lzybkr Thanks for the review and useful comments. 👍

**Pull request merged**